### ISSUE TITLE
1.8 azure-iot-device does not complie with typesciprt

### ISSUE THREAD

**Issue by StefanSchoof** [Azure/azure-iot-sdk-node#403]
# Context
- **OS and version used:** Windows 10
- **Node.js version:** v8.12.0
- **npm version:** 6.4.1
- **list of installed packages:** see below
- **cloned repo:** N/A

# Description of the issue:
With the new 1.8 release of azure-iot-device typescript throws an error in the module.

# Code sample exhibiting the issue:
Create a index.ts in an empty folder with the content:

```
import { Client } from 'azure-iot-device';
```

run
1. `npm install azure-iot-device@1.7 typescript @types/node`
2. `npx tsc --init`
3. `npx tsc` (runs succesfully)
4. `npm install azure-iot-device@1.8`
5. `npx tsc` 
 got the error:
```
node_modules/azure-iot-device/lib/sas_authentication_provider.d.ts:30:5 - error TS2416: Property 'getDeviceCredentials' in type 'SharedAccessSignatureAuthenticationProvider' is not assignable to the same property in base type 'AuthenticationProvider'.
  Type '(callback?: DoubleValueCallback<Error, TransportConfig> | undefined) => void | Promise<TransportConfig>' is not assignable to type '(callback: (err: Error, credentials?: TransportConfig | undefined) => void) => void'.
    Types of parameters 'callback' and 'callback' are incompatible.
      Type '(err: Error, credentials?: TransportConfig | undefined) => void' is not assignable to type 'DoubleValueCallback<Error, TransportConfig>'.

30     getDeviceCredentials(callback?: Callback<TransportConfig>): Promise<TransportConfig> | void;
       ~~~~~~~~~~~~~~~~~~~~

node_modules/azure-iot-device/lib/x509_authentication_provider.d.ts:22:5 - error TS2416: Property 'getDeviceCredentials' in type 'X509AuthenticationProvider'
is not assignable to the same property in base type 'AuthenticationProvider'.
  Type '(callback?: DoubleValueCallback<Error, TransportConfig> | undefined) => void | Promise<TransportConfig>' is not assignable to type '(callback: (err: Error, credentials?: TransportConfig | undefined) => void) => void'.
    Types of parameters 'callback' and 'callback' are incompatible.
      Type '(err: Error, credentials?: TransportConfig | undefined) => void' is not assignable to type 'DoubleValueCallback<Error, TransportConfig>'.
        Types of parameters 'err' and 'result1' are incompatible.
          Type 'Error | undefined' is not assignable to type 'Error'.
            Type 'undefined' is not assignable to type 'Error'.

22     getDeviceCredentials(callback?: Callback<TransportConfig>): Promise<TransportConfig> | void;
```

# Console log of the issue:

**Comment by anthonyvercolano** [comment]
Fixed with #405

**Comment by az-iot-builder-01** [comment]
@StefanSchoof, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
